libraries/AP_Quicktune/AP_Quicktune.cpp

```diff
@@ -0,0 +1,672 @@
+            return &attitude_control.get_rate_roll_pid().ff();
+        case param_s::PIT_P:
+            return &attitude_control.get_rate_pitch_pid().kP();
+        case param_s::PIT_I:
+            return &attitude_control.get_rate_pitch_pid().kI();
+        case param_s::PIT_D:
+            return &attitude_control.get_rate_pitch_pid().kD();
+        case param_s::PIT_SMAX:
+            return &attitude_control.get_rate_pitch_pid().slew_limit();
+        case param_s::PIT_FLTT:
+            return &attitude_control.get_rate_pitch_pid().filt_T_hz();
+        case param_s::PIT_FLTD:
+            return &attitude_control.get_rate_pitch_pid().filt_D_hz();
+        case param_s::PIT_FLTE:
+            return &attitude_control.get_rate_pitch_pid().filt_E_hz();
+        case param_s::PIT_FF:
+            return &attitude_control.get_rate_pitch_pid().ff();
+        case param_s::YAW_P:
+            return &attitude_control.get_rate_yaw_pid().kP();
+        case param_s::YAW_I:
+            return &attitude_control.get_rate_yaw_pid().kI();
+        case param_s::YAW_D:
+            return &attitude_control.get_rate_yaw_pid().kD();
+        case param_s::YAW_SMAX:
+            return &attitude_control.get_rate_yaw_pid().slew_limit();    
+        case param_s::YAW_FLTT:
+            return &attitude_control.get_rate_yaw_pid().filt_T_hz();
+        case param_s::YAW_FLTD:
+            return &attitude_control.get_rate_yaw_pid().filt_D_hz();
+        case param_s::YAW_FLTE:
+            return &attitude_control.get_rate_yaw_pid().filt_E_hz();
+        case param_s::YAW_FF:
+            return &attitude_control.get_rate_roll_pid().ff();
+        default:
+            GCS_SEND_TEXT(MAV_SEVERITY_EMERGENCY, "INTERNAL ERROR - get_param_pointer - param was %d", uint8_t(param));
+            // INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+            return nullptr;
+    }
+}
+
+float AP_Quicktune::get_param_value(AP_Quicktune::param_s param){
+    AP_Float *ptr = get_param_pointer(param);
+    if (ptr != nullptr) {
+        return ptr->get();
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_EMERGENCY, "INTERNAL ERROR - get_param_value - param was %d", uint8_t(param));
+    return 0.0;
+}
+
+void AP_Quicktune::set_param_value(AP_Quicktune::param_s param, float value){
+    AP_Float *ptr = get_param_pointer(param);
+    if (ptr != nullptr) {
+       ptr->set(value);
+       return;
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_EMERGENCY, "INTERNAL ERROR - set_param_value - param was %d", uint8_t(param));
+    return;
+}
+
+void AP_Quicktune::set_and_save_param_value(AP_Quicktune::param_s param, float value){
+    AP_Float *ptr = get_param_pointer(param);
+    if (ptr != nullptr) {
+       ptr->set_and_save(value);
+       return;
+    }
+    GCS_SEND_TEXT(MAV_SEVERITY_EMERGENCY, "INTERNAL ERROR - set_param_value - param was %d", uint8_t(param));
+    return;
+}
+
+AP_Quicktune::axis_names AP_Quicktune::get_axis(AP_Quicktune::param_s param)
```

Comment:
```suggestion
AP_Quicktune::axis_names AP_Quicktune::get_axis(AP_Quicktune::param_s param) const
```

.... and elsewhere.  Generally `const` as much as possible, it prevents bugs and is one of the few guardrails the language gives you!

Reply:
done